assets/src/edit-story/components/panels/link.js

```diff
@@ -0,0 +1,205 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { rgba } from 'polished';
+import { debounce } from 'throttle-debounce';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useCallback } from 'react';
+import { Input as BaseInput, Row } from '../form';
+import { createLink } from '../link';
+import { ReactComponent as Close } from '../../icons/close_icon.svg';
+import { SimplePanel } from './panel';
+import getCommonValue from './utils/getCommonValue';
+
+const Note = styled.span`
+  color: ${({ theme }) => theme.colors.mg.v1};
+  font-size: 11px;
+  line-height: 16px;
+  margin-bottom: 6px;
+`;
+
+/** TODO(@wassgha): Replace with text input component once done */
+const Input = styled(BaseInput)`
+  width: 100%;
+  border: none !important;
+  margin: 0;
+  border-radius: 4px !important;
+`;
+
+/** TODO(@wassgha): Replace with text input component once done */
+const InputContainer = styled.div`
+  position: relative;
+  width: 100%;
+  background: ${({ theme, disabled }) =>
+    disabled ? theme.colors.fg.v3 : theme.colors.fg.v1};
+  color: ${({ theme }) => theme.colors.mg.v1};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  border: 1px solid ${({ theme }) => theme.colors.fg.v3} !important;
+  border-radius: 4px !important;
+  box-sizing: border-box;
+  display: flex;
+  align-items: center;
+`;
+
+const BrandIcon = styled.img`
+  width: 64px;
+  height: 64px;
+  border-radius: 50%;
+  background-color: ${({ theme }) => theme.colors.fg.v3};
+  border: none;
+  margin-right: 8px;
+`;
+
+const DeleteButton = styled.button`
+  position: absolute;
+  right: 8px;
+  appearance: none;
+  background: ${({ theme }) => rgba(theme.colors.fg.v0, 0.54)};
+  width: 16px;
+  height: 16px;
+  border: none;
+  padding: 0px;
+  border-radius: 50%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const CloseIcon = styled(Close)`
+  width: 12px;
+  height: 12px;
+`;
+
+function LinkPanel({ selectedElements, onSetProperties }) {
+  const link = getCommonValue(selectedElements, 'link') || null;
+  const isFill = getCommonValue(selectedElements, 'isFill');
+  const [state, setState] = useState({ link: createLink() });
+  useEffect(() => {
+    setState({ link });
+  }, [link]);
+  const handleSubmit = (evt) => {
+    onSetProperties(state);
+    evt.preventDefault();
+  };
+  const canLink = selectedElements.length === 1 && !isFill;
+  const populateMetadata = useCallback(
+    debounce(300, async (/** url */) => {
```

Comment:
When doing this in a `useEffect` hook, we could probably just use `setTimeout` instead of `debounce`. Thoughts?

Reply:
Can you elaborate on why `setTimeout` is better than debouncing here?

Reply:
I was just wondering/curious as a new dependency for this might be unneeded when using hooks. Also, we‘re gonna have similar scenarios for things like delayed video poster extraction, where we could try to find a more generally applicable solution.

Reply:
I guess I'm just confused as to how hooks affect this, we still want to essentially only make *one* request after the user finishes typing, `setTimeout` alone doesn't do this and anything else is going to be essentially a re-implementation of `debounce` imho

Reply:
Fair enough. I guess I was just thinking out loud :-)